benchmarker/modules/i_neural_net.py

```diff
@@ -41,23 +39,25 @@ def __init__(self, params, extra_args=None):
                 self.params["batch_size_per_device"] * self.params["nb_gpus"]
             )
         self.params["channels_first"] = True
-        path = f"benchmarker.modules.problems.{params['problem']['name']}.{params['framework']}"
-        kernel_module = importlib.import_module(path)
         if parsed_args.random_seed is not None:
             self.set_random_seed(int(parsed_args.random_seed))
-        self.get_kernel(kernel_module, remaining_args)
+        self.get_kernel(params, remaining_args)
 
-    def get_kernel(self, module, remaining_args):
+    def get_kernel(self, params, remaining_args):
         """Default function to set `self.net`.  The derived do_* classes can
         override this function if there is some framework specific
         logic involved (e.g. GPU/TPU management etc).
-
-        :param module: module which implements the framework specific
-        `get_kernel` method.
-        :param remaining_args: (Sub)problem dependent arguments.
-
         """
-        self.net = module.get_kernel(self.params, remaining_args)
+        path_params = f"benchmarker.modules.problems.{params['problem']['name']}.params"
+        path_kernel = (f"benchmarker.modules.problems.{params['problem']['name']}."
+                       f"{params['framework']}")
+        module_kernel = importlib.import_module(path_kernel)
+        try:
```

Comment:
on ugly - could not find better way to check if module is available
you can suggest

yes, extra args are saved in params, but so far some of the submodules are doing this differently, so I did not yet move to a single place
this will be in the next iteration